test/pkg.jl

```diff
@@ -367,7 +367,10 @@ temp_pkg_dir() do project_path
         tasks = Task[]
         iobs = IOBuffer[]
         Sys.CPU_THREADS == 1 && error("Cannot test for atomic usage log file interaction effectively with only Sys.CPU_THREADS=1")
-        run(`$(Base.julia_cmd()) --project="$(pkgdir(Pkg))" -e "import Pkg"`) # to precompile Pkg given we're in a different depot
+        # to precompile Pkg given we're in a different depot
+        run(`$(Base.julia_cmd()) --project="$(pkgdir(Pkg))" -e "import Pkg"`)
+        # make sure the General registry is installed
+        run(pipeline(`$(Base.julia_cmd()) --project="$(pkgdir(Pkg))" -e "import Pkg; Pkg.Registry.add()"`; stdout=devnull, stderr=devnull))
```

Comment:
Piping stderr to null is usually a bad idea, since that is where fatal error messages go

Reply:
I don't want to show the output unless there is an error.

How about I pipe `stderr` to an `IOBuffer`, and print the contents of the `IOBuffer` only if the command fails?

Reply:
stderr is where errors go to, so that should be the same

Reply:
> stderr is where errors go to, so that should be the same

Unfortunately, it looks like `Pkg` is violating that rule 😂 . It prints the `Installing known registries into...` message to `stderr` even if there are no errors.

To reproduce:

```
julia> env2 = copy(ENV);

julia> delete!(env2, "JULIA_PKG_SERVER");

julia> env2["JULIA_DEPOT_PATH"] = mktempdir();

julia> julia = Base.julia_cmd().exec[1];

julia> cmd = `$(julia) -e 'import Pkg; Pkg.Registry.add(Pkg.Registry.RegistrySpec[])'`;

julia> run(pipeline(setenv(cmd, env2); stdout=devnull));
  Installing known registries into `/var/folders/dg/7mtlrlqd6fz01jdb9djb33bm0000gp/T/jl_NitKkG`
```

I've redirected `stdout` to `devnull`, so I'm assuming that the `Installing known registries...` message is being printed to `stderr`.

Reply:
Alright, here's what I just pushed:
1. We don't redirect `stderr`, so `stderr` will always be shown (regardless of the command's exit code).
2. `stdout` will be shown if and only if the command exits with nonzero exit code.